libdevcore/CommonData.h

```diff
@@ -159,6 +161,17 @@ inline std::string toCompactBigEndianString(_T _val)
 	return ret;
 }
 
+/// Convenience function for conversion of a u256 to hex
+inline std::string toHex(u256 val, HexPrefix prefix = HexPrefix::DontAdd)
+{
+	std::string str = toHex(toBigEndian(val));
+	return (prefix == HexPrefix::Add) ? "0x" + str : str;
+}
+inline std::string toCompactHex(u256 val, HexPrefix prefix = HexPrefix::DontAdd)
+{
+	std::string str = toHex(toCompactBigEndian(val, 1)); //1 means val=0 would be '0x00' instead of '0x'
```

Comment:
it should not. toHex at line 54  accept    _T const& _data
the issues was in makeAllFieldHex function. 
data fields are ok